cmd/query/main.go

```diff
@@ -141,11 +144,36 @@ func main() {
 			compressHandler := handlers.CompressHandler(r)
 			recoveryHandler := recoveryhandler.NewRecoveryHandler(logger, true)
 
+			// Create GRPC Server.
+			grpcServer := grpc.NewServer()
+
+			// Add logging.
+			grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, os.Stderr, os.Stderr))
+
+			// Register GRPC Handler.
+			spanReaderGRPC, err := storageFactory.CreateSpanReader()
+			if err != nil {
+				logger.Fatal("Failed to create span reader", zap.Error(err))
+			}
+			spanReaderGRPC = storageMetrics.NewReadMetricsDecorator(spanReaderGRPC, baseFactory.Namespace(metrics.NSOptions{Name: "query-grpc", Tags: nil}))
+			dependencyReaderGRPC, err := storageFactory.CreateDependencyReader()
```

Comment:
Should we be creating different instances of the `spanReader` and `dependencyReader` for use with GRPC? I thought to do so in the case that a user might want to use both HTTP and GRPC endpoints at the same time (possibly HTTP for the UI and a custom GRPC querier), but I'm not so sure that's required.

Reply:
Should not be creating any storage components, they are already passed to QueryService.